nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -272,6 +279,38 @@ void ObstacleLayer::onInitialize()
   }
 }
 
+rcl_interfaces::msg::SetParametersResult
+ObstacleLayer::dynamicParametersCallback(
+  std::vector<rclcpp::Parameter> parameters)
+{
+  std::lock_guard<Costmap2D::mutex_t> guard(*getMutex());
```

Comment:
I think you need to add `   std::lock_guard<Costmap2D::mutex_t> guard(*getMutex());` at the start of update bounds

Reply:
I've moved the mutex locking from updateFootprint to updateBounds so that lock_guard isn't called twice.